lib/matplotlib/axes/_base.py

```diff
@@ -2059,7 +2059,8 @@ def draw(self, renderer=None, inframe=False):
         if not self.get_visible():
             return
         renderer.open_group('axes')
-
+        # prevent triggering call backs during the draw process
+        self._stale = True
```

Comment:
This can probably be removed now that stale state _always_ propagates (which I am not 100% happy with) and with the context manager in `draw_idle`.  As I babbled in the comments, the stale code is very aggressive about setting artists to be stale (even if it hasn't changed).  In the case of `Agg` this was causing redraws to be trigged from inside of a draw.